import "goby/acomms/protobuf/modem_message.proto";

package goby.acomms.protobuf;

message MTDataRequest
{
  required int32 modem_id = 1;
  repeated goby.acomms.protobuf.ModemTransmission outbox = 2;
  optional uint64 request_id = 3;

  enum Command
  { 
    SEND_BYE = 1;           
    HANGUP = 2;
  }
  optional Command command = 4;
}

message MTDataResponse
{
  required bool mobile_node_connected = 1;
  required int32 modem_id = 2;
  optional uint64 request_id = 4;
}

message MODataAsyncReceive
{
  required int32 modem_id = 1;
  required goby.acomms.protobuf.ModemTransmission inbox = 2;
}
